src/utils.rs

```diff
@@ -0,0 +1,107 @@
+use std::cmp::Ordering;
+use std::cmp::Ordering::{Less, Equal, Greater};
+
+/// Takes two *sorted* vectors and a comparison function
+/// Gives back a tuple of vectors:
+///  - one for the elements unique to the first vector
+///  - one for the pairs of elements found equal
+///  - one of the elements unique to the second vector
+pub fn find_overlap<F, T>(mut left: Vec<T>,
+                          mut right: Vec<T>,
+                          mut fun: F)
+                          -> (Vec<T>, Vec<(T, T)>, Vec<T>)
```

Comment:
I feel like returning a struct instead of a tuple here would probably make this function signature easier to read.